string id
int32 val0
float64 val1
float64 val2
